### PR TITLE
Make karavi-topology listening port configurable

### PR DESCRIPTION
**Pull request by ameer-jabbar**
# Description

Make the listening port that the karavi-topology service binds to configurable. OpenShift deployment needs to use higher number port because pods run unprivileged. 

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|    https://github.com/dell/helm-charts/issues/54    |

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have run 'make check' to ensure my code does not have any formatting, vetting, linting, or security issues
- [x] I have run 'make test' to ensure new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degrade
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Manual test with Openshift deployment. Create an image, deployed to internal docker repo, referenced the image in values.yaml, ran helm deployment. Then ran manual e2e test.

### CODE REVIEW

**Comment by ameer-jabbar** [comment]
Run e2e

**Comment by ameer-jabbar** [comment]
Run e2e test

**Comment by ameer-jabbar** [comment]
Run e2e test

**Comment by ameer-jabbar** [comment]
Run e2e test

**Comment by ameer-jabbar** [comment]
Run e2e test

**Comment by ameer-jabbar** [comment]
e2e pass

File: cmd/topology/main.go [hunk 1]
```diff
@@ -50,10 +51,21 @@ func main() {
 		keyFile = defaultKeyFile
 	}
 
+	var bindPort int
+	portEnv := os.Getenv("PORT")
+	if portEnv != "" {
```
Comment (hoppea2) [inline cmd/topology/main.go:56]: What is the port value if PORT=""

Comment (ameer-jabbar) [reply cmd/topology/main.go:56]: In that case, bindPort should be init'ed to 0 and in internal/service/service.go, this value means to use the default, which would be 443.

**Review by hoppea2 — APPROVED** [review]

**Review by tdawe — APPROVED** [review]

**Review by tdawe — DISMISSED** [review]

**Comment by ameer-jabbar** [comment]
e2e pass

**Review by hoppea2 — APPROVED** [review]

**Review by atye — APPROVED** [review]

**Pull request closed without merging**